internal/controllers/core/podlogstream/podlogstreamcontroller.go

```diff
@@ -131,53 +132,69 @@ func (m *Controller) shouldStreamContainerLogs(c v1alpha1.Container, key podLogK
 }
 
 // Reconcile the given stream against what we're currently tracking.
-func (r *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
 	stream := &PodLogStream{}
 	streamName := req.NamespacedName
-	err := r.client.Get(ctx, req.NamespacedName, stream)
-	r.indexer.OnReconcile(req.NamespacedName, stream)
+	err := c.client.Get(ctx, req.NamespacedName, stream)
+	c.indexer.OnReconcile(req.NamespacedName, stream)
 	if apierrors.IsNotFound(err) {
-		r.podSource.handleReconcileRequest(ctx, req.NamespacedName, stream)
-		r.deleteStreams(streamName)
+		_ = c.podSource.handleReconcileRequest(ctx, req.NamespacedName, stream)
```

Comment:
Nit: I'm always suspicious of ignored errors, so a comment here would be nice since it's not immediately obvious that the error is meant for attaching to the object that no longer exists and is impossible to even hit (alternatively, a more explicit `PodSource::Remove()` type method)

Reply:
done